### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by menyagah**
# In this milestone: 3
- I removed all hardcoded items from the tasks array.
- I created a new JavaScript file for the new functionality.
- I implemented a function for adding a new task (add a new element to the array).
- I implemented a function for deleting a task (remove an element from the array).
-  I implemented a function for the "Clear all completed" button (use filter() method).
- I implemented the delete task functionality
- I implemented the edit task functionality. (DOUBLE CLICK ON A TASK TO EDIT IT)
- Ensured that all changes to the To-Do List are being saved in local storage.
-  Below is a direct link to the live version
    https://raw.githack.com/menyagah/ToDoList-App/feature/dist/index.html
- Fix Tse required changes

### CODE REVIEW

File: src/addItem.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+const addTodo = () => {
+  const input = document.querySelector('.input-field');
+  input.addEventListener('keypress', (e) => {
+    if (e.key === 'Enter') {
+      if (input.value === '') {
+        // eslint-disable-next-line no-alert
+        alert('You cannot add empty fields');
+      } else {
+        const todos = JSON.parse(localStorage.getItem('todo') || '[]');
+        const id = Math.random().toString(36).substr(0, 5);
```
Comment (rloterh) [inline src/addItem.js]: - [ ] New entries **do not match** the property index. The _property index_ must be set to the value of the _new array length_ (i.e. if a 5th task is added to the list, the index of that task should equal 5).

**Index mismatch in your application:**
image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Martin, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Good Points 👍 
- CRUD feature is functioning as intended
- Functions effectively split into various files
- Descriptive Pull Request
- Good ReadMe
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/addItem.js (hunk 1 above)
Comment (menyagah) [reply src/addItem.js]: Hi @rloterh , I am populating an index based on a random number and converting it to a string but I will populate my id as you have recommended based on the array length.

Comment (rloterh) [reply src/addItem.js]: This issue is well resolved! ✔️

Comment (menyagah) [reply src/addItem.js]: Yeey! Thank you for the timely review!

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Martin, 

Your project is complete! Great job for making the necessary changes recommended by the previous reviewer(s). There is nothing else to say other than... it's time to merge it 🍾:shipit:.
Congratulations! 🎉🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by menyagah** [comment]
Thank you @rloterh .

**Pull request merged**